### ISSUE TITLE
Setting ContentType property on the brokered message might come handy for Azure Functions interop

### ISSUE THREAD

**Issue by danielmarbach** [Particular/NServiceBus.Transport.AzureServiceBus#19]
Azure Functions provides POCO support for XML and JSON. Meaning it is possible to define a function in the following way:

```
public static class MyFunctions
    {
        [FunctionName("MyFunction")]
        public static void Run([ServiceBusTrigger("myfunctionqueue", AccessRights.Manage, Connection = "AzureServiceBusConnectionString")]MyPayload payload, TraceWriter log)
        {
        }
}
```

Function runtime makes the following assumptions:

- If `ContentType` is set to `application/json` it deserializes it with Newtonsoft JSON (I think it is version 9 in function runtime 1.x)
- If `ContentType` is set to `application/xml` it tries to deserialize it with the data contract deserializer assuming it being the data contract xml format
- If `ContentType` is not set then it tries to deserialize it with the data contract deserializer (which most of the time fails)

In theory, we could promote `NServiceBus.ContenType` header to `ContentType` property of the brokered message. This would then allow people that are using the Newtonsoft Json serializer (which is the recommended default for ASB, ASQ...) to do seamless "interop" with functions POCO support. Unfortunately, if the core XML Serializer is used it would send `application/xml` and then functions would try to use the data contract deserializer to read that xml but still fail because the data contract xml format has different requirements.

Core XML

```
<?xml version="1.0" encoding="utf-8"?>
<MyEvent xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns="http://tempuri.net/">
  <Property>Event from Endpoint1</Property>
</MyEvent>
```

vs

DataContract XML
```
<?xml version="1.0" encoding="utf-8"?>
<MyEvent xmlns:i="http://www.w3.org/2001/XMLSchema-instance" xmlns="http://schemas.datacontract.org/2004/07/">
  <Property>Event from Endpoint1</Property>
</MyEvent>
```

So as you can see even if we did set the ContentType it would only make the interoperability story better for people who are using the Newtonsoft Serializer but at least it would make it better for them. There are a few more caveats that make azure functions a bit weird in terms of assumptions. When you use the collector API with POCO the output serialization is assumed to be JSON. It is possible though to support converters if necessary

https://github.com/Azure/azure-webjobs-sdk/wiki/Creating-custom-input-and-output-bindings

It might make this discussion to steer away from ASB towards a more platform-centric approach towards functions which even might be the better move. Anyway, I'm still raising this here for completeness reason.

**Comment by antongeorgiev** [comment]
Implemented in #47

**Comment by SeanFeldman** [comment]
Fixed via https://github.com/Particular/NServiceBus.Transport.AzureServiceBus/commit/bc06995a69232b7965e15bee029107a2a6ffa876

Thank you @antongeorgiev 🙂
